cmds/replay.c

```diff
@@ -816,6 +818,14 @@ static int print_graph_rstack(struct uftrace_data *handle,
 		}
 	}
 
+	if (opts->srcline) {
+		loc = task_find_loc_addr(sessions, task, rstack->time, rstack->addr);
+		if (loc) {
+			str_loc = xmalloc(strlen(loc->file->name) + 11); // 11: length of integer max value + NULL
+			sprintf(str_loc, "[%s:%d]", loc->file->name, loc->line);
```

Comment:
Please use snprintf() instead.

Reply:
I tried to use it. But length value (second argument) for snprintf is same with size of str_loc (first argument). so compiler give warning for that.

Reply:
Hmm.. can you show me the warning?

Reply:
If you have a smart compiler.. maybe you can do like this..

```c
/* comments here */
srcline_len = strlen(loc->file->name) + 10
str_loc = xmalloc(srcline_len + 1);
snprintf(str_loc, srcline_len, ...)
str_loc[srcline_len] = '\0';
```

Reply:
I think `xasprintf` might be enough.
```c
xasprintf(&str_loc, "%s:%d", loc->file->name, loc->line);
```